src/main/java/org/sagebionetworks/bridge/models/schedules/ActivityScheduler.java

```diff
@@ -162,7 +162,7 @@ private LocalDateTime getExpiresOn(LocalDate localDate, LocalTime localTime) {
             Iterable<String> eventIds = Schedule.EVENT_ID_SPLITTER.split(eventIdsString.trim());
             for (String thisEventId : eventIds) {
                 if (context.getEvent(thisEventId) != null) {
-                    eventDateTimeList.add(context.getEvent(thisEventId));
+                    eventDateTimeList.add(context.getEvent(thisEventId).withZone(context.getRequestTimeZone()));
```

Comment:
Is it worth adding a unit test that calls getEventDateTimes() directly, to ensure all datetimes returned by this method are in the correct time zone?

Reply:
Added a test that the time zone conversion happens in this method.